tests/components/zone/test_init.py

```diff
@@ -1,229 +1,224 @@
 """Test zone component."""
-
-import unittest
-from unittest.mock import Mock
+from asynctest import patch
+import pytest
 
 from homeassistant import setup
 from homeassistant.components import zone
-
-from tests.common import MockConfigEntry, get_test_home_assistant
-
-
-async def test_setup_entry_successful(hass):
-    """Test setup entry is successful."""
-    entry = Mock()
-    entry.data = {
-        zone.CONF_NAME: "Test Zone",
-        zone.CONF_LATITUDE: 1.1,
-        zone.CONF_LONGITUDE: -2.2,
-        zone.CONF_RADIUS: True,
-    }
-    hass.data[zone.DOMAIN] = {}
-    assert await zone.async_setup_entry(hass, entry) is True
-    assert "test_zone" in hass.data[zone.DOMAIN]
-
-
-async def test_unload_entry_successful(hass):
-    """Test unload entry is successful."""
-    entry = Mock()
-    entry.data = {
-        zone.CONF_NAME: "Test Zone",
-        zone.CONF_LATITUDE: 1.1,
-        zone.CONF_LONGITUDE: -2.2,
+from homeassistant.components.zone import DOMAIN
+from homeassistant.const import (
+    ATTR_EDITABLE,
+    ATTR_FRIENDLY_NAME,
+    ATTR_ICON,
+    ATTR_NAME,
+    SERVICE_RELOAD,
+)
+from homeassistant.core import Context
+from homeassistant.exceptions import Unauthorized
+from homeassistant.helpers import entity_registry
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def storage_setup(hass, hass_storage):
+    """Storage setup."""
+
+    async def _storage(items=None, config=None):
+        if items is None:
+            hass_storage[DOMAIN] = {
+                "key": DOMAIN,
+                "version": 1,
+                "data": {
+                    "items": [
+                        {
+                            "id": "from_storage",
+                            "name": "from storage",
+                            "latitude": 1,
+                            "longitude": 2,
+                            "radius": 3,
+                            "passive": False,
+                            "icon": "mdi:from-storage",
+                        }
+                    ]
+                },
+            }
+        else:
+            hass_storage[DOMAIN] = {
+                "key": DOMAIN,
+                "version": 1,
+                "data": {"items": items},
+            }
+        if config is None:
+            config = {}
+        return await setup.async_setup_component(hass, DOMAIN, config)
+
+    return _storage
+
+
+async def test_setup_no_zones_still_adds_home_zone(hass):
+    """Test if no config is passed in we still get the home zone."""
+    assert await setup.async_setup_component(hass, zone.DOMAIN, {"zone": None})
+    assert len(hass.states.async_entity_ids("zone")) == 1
+    state = hass.states.get("zone.home")
+    assert hass.config.location_name == state.name
+    assert hass.config.latitude == state.attributes["latitude"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4870504)